c7n/resources/airflow.py

```diff
@@ -113,3 +115,67 @@ def process_resource_set(self, client, airflow, tags):
 
 ApacheAirflow.filter_registry.register('marked-for-op', TagActionFilter)
 ApacheAirflow.action_registry.register('mark-for-op', TagDelayedAction)
+
+@ApacheAirflow.action_registry.register('update-environment')
+class UpdateApacheAirflowEnvironment(Action):
+    """
+    Action to update an Airflow environment to
+    set the WebserverAccessMode to PRIVATE_ONLY or PUBLIC_ONLY.
+
+    :example:
+
+    .. code-block:: yaml
+
+        policies:
+          - name: set-webserver-access-mode
+            resource: airflow
+            actions:
+              - type: update-environment
+                access_mode: PRIVATE_ONLY
+    """
+
+    permissions = ('airflow:UpdateEnvironment',)
+
+    schema = type_schema(
+        'update-environment',
+        access_mode={'type': 'string', 'enum': ['PRIVATE_ONLY', 'PUBLIC_ONLY']},
+        required=['access_mode']
+    )
+
+    def process(self, resources):
```

Comment:
I think this should filter out all the resources in `Status: CREATE_FAILED` because apparently the resources in create failed status cannot be updated. Trackeback:
```
botocore.errorfactory.ValidationException: An error occurred (ValidationException) when calling the UpdateEnvironment operation: Environments with CREATE_FAILED status cannot be updated.
```

Reply:
Good catch, will add. Thanks!